### DISCUSSION TITLE
【Question】Face upward in ragdoll may cause the actor rotation change in one frame

### DISCUSSION THREAD

**Discussion by watsonsong** [Sixze/ALS-Refactored#393] [Q&A] [answered]
When ragdoll enable, it check whether face upward by pelvis roll each frame. And if the face direction changed, it change the actor rotation 180 degree in one frame.
When the change happend, it will produce a huge velocity, and make things like cloth or spring bones work wrong.
```
RagdollingState.bFacedUpward = PelvisRotation.Roll <= 0.0f;

auto NewActorRotation{GetActorRotation()};
NewActorRotation.Yaw = RagdollingState.bFacedUpward ? PelvisRotation.Yaw - 180.0f : PelvisRotation.Yaw;

SetActorLocationAndRotation(NewActorLocation, NewActorRotation);
```

Maybe it could be better use ETeleportType::TeleportPhysics to keep velocity from a instant large value?
Or just not set rotation during every tick, just set the rotation when stop ragdolling and starting to get up?

**Comment by Sixze** [comment, marked as answer]
`ETeleportType::TeleportPhysics` is not an option, because it also teleports the ragdoll's physical bodies, while without it they "'kinda'" all remain unchanged. "Kinda" because actually, for some unknown reason, changing the actor's transform without `ETeleportType::TeleportPhysics` still affects the ragdoll a bit.

I have recently reworked ragdolling, now when it is activated the character's mesh detaches from the capsule and remains fixed in place, so changing the actor transform does not affect the mesh in any way now.
